### PR TITLE
Finish implementing c.DONATION_SLIDER_DISPLAY

### PR DESCRIPTION
**Pull request by kitsuta**
Makes amount_extra actually work when this option is turned off.

### CODE REVIEW

File: uber/templates/base.html [hunk 1]
```diff
@@ -165,7 +165,10 @@
     {% block head_additional %}{% endblock %}
 </head>
 <body>
-    {% if c.PRE_CON %}<div class="loader"><a class="loader_link" href="../static_views/slow_load.html" target="_blank"></a></div>{% endif %}
+    {% if c.PRE_CON %}
```
Comment (kitsuta) [inline uber/templates/base.html]: This is an unrelated change that makes `COLLECT_FULL_ADDRESS` work.

**Comment by kitsuta** [comment]
I've also (mostly accidentally) included some other fixes needed by Anthrocon.

File: uber/model_checks.py [hunk 2]
```diff
@@ -120,7 +120,7 @@ def emergency_contact(attendee):
 @validation.Attendee
 @ignore_unassigned_and_placeholders
 def cellphone(attendee):
-    if attendee.cellphone and _invalid_phone_number(attendee.cellphone):
+    if attendee.cellphone and _invalid_phone_number(attendee.cellphone) and not c.COLLECT_FULL_ADDRESS:
```
Comment (EliAndrewC) [inline uber/model_checks.py]: See my comment on your other PR that (accidentally?) included these changes:
https://github.com/magfest/ubersystem/pull/1566/files#r42947566

File: uber/templates/base.html [hunk 3]
```diff
@@ -165,7 +165,10 @@
...
+        <div class="loader"><a class="loader_link" href="../static_views/slow_load.html" target="_blank"></a></div>
+        <script src="https://maps.googleapis.com/maps/api/js?sensor=false&amp;libraries=places"></script>
```
Comment (EliAndrewC) [inline uber/templates/base.html]: The ``&amp;`` is almost certainly not what you want here, and may adversely affect what gets loaded.

File: uber/templates/preregistration/preregbase.html [hunk 4]
```diff
@@ -129,7 +129,12 @@
         var setKickinFromBadge = function (types, index) {
             var type = types.options[index];
             if (type.extra !== undefined && type.extra !== null && $.field('amount_extra')) {
-                $('input:radio[name=amount_extra][value='+ type.extra +']').prop('checked', true).trigger('change');
+                {% if c.DONATION_SLIDER_DISPLAY %}
+                    $('input:radio[name=amount_extra][value='+ type.extra +']').prop('checked', true).trigger('change');
+                {% else %}
+                    $.field('amount_extra').val(type.extra);
+                    $.field('amount_extra').trigger('change');
+                {% endif %}
```
Comment (EliAndrewC) [inline uber/templates/preregistration/preregbase.html]: +1 on this fix.

File: uber/templates/regform.html [hunk 5]
```diff
@@ -52,8 +52,10 @@
 
     {% if not attendee.is_new %}
         // This removes any badge levels lower than the attendee has purchased already
-        while (_(BADGE_TYPES.options).size() && BADGE_TYPES.options[0].extra < {{ attendee.amount_extra }}) {
-            BADGE_TYPES.options.splice(0, 1);
```
Comment (EliAndrewC) [inline uber/templates/regform.html]: Some browsers will still raise an error if you refer to a variable which has not been defined, even when just checking its type to see if it's undefined!  However, the way around this is to go through ``window``.  In case you're not aware, ``x = 5`` is the same as saying ``window.x = 5`` when running Javascript in the browser.  But it's always safe to say ``window.foo`` since that will never raise an exception, and if ``foo`` was never defined it just returns ``undefined``, whereas some browsers will still give you an exception if you say ``foo`` without having defined ``foo`` first.

A better way to write this if statement would therefore be to just say ``if (window.BADGE_TYPES)``, which also checks for ``null``, which is good because we'd get an exception if we tried to enter this if-block when ``BADGE_TYPES`` was null.  (Not that it ever should be, but defensive coding and all that.)

**Comment by EliAndrewC** [comment]
+1 on the fixes, but see my comments on phone number validation and the ``&amp;`` typo before merging.

**Comment by kitsuta** [comment]
Finally got around to implementing the fixes, so I'm merging this now.

**Comment by EliAndrewC** [comment]
+1 on the fixes
